### PR TITLE
test: cri-containerd test

### PR DESCRIPTION
**Pull request by jcvenegas**
Add initial support for cri-containerd tests.

### CODE REVIEW

**Comment by jcvenegas** [comment]
@grahamwhaley  @chavafg I got an error from metrics ci, I suspect is the golang version. 

```
 vendor/github.com/stevvooe/ttrpc/unixcreds_linux.go:21: uc.SyscallConn undefined (type *net.UnixConn has no field or method SyscallConn)
```
Where is installed golang in metrics ci ?

**Comment by grahamwhaley** [comment]
@jcvenegas that is (or should be) v1.8.3 of go on the metrics CI build slaves.

**Comment by grahamwhaley** [comment]
it is installed in the standard go path `/usr/local/go`. We set that up in the Jenkins config file/dialog:

```
export PATH=/usr/local/go/bin:${GOPATH}/bin:/sbin:${PATH}
```

And you can find in the logs somewhere around the auto-code generation section:

```
11:40:00 • golang:
11:40:00 	go version go1.8.3 linux/amd64
```

File: .ci/install_cri-containerd.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+cidir=$(dirname "$0")
+source "${cidir}/lib.sh"
+get_cc_versions
+# Fixme add a version to use in versions.txt file
```
Comment (jodh-intel) [inline .ci/install_cri-containerd.sh]: Is there an issue for this, or are you planning to update this PR to remove the "Fixme"?

File: .ci/setup_env_fedora.sh [hunk 2]
```diff
@@ -50,6 +50,9 @@ chronic sudo -E dnf -y install btrfs-progs-devel device-mapper-devel 	  \
 	libgpg-error-devel libseccomp-devel libselinux-devel ostree-devel \
 	pkgconfig go-md2man
 
+echo "Install cri-containerd dependencies"
+chronic sudo -E dnf -y install libseccomp-devel btrfs-progs-devel libseccomp-static
```
Comment (jodh-intel) [inline .ci/setup_env_fedora.sh]: OOI, why do we need the `-static` version of this library? It doesn't appear to be needed on Ubuntu too.

File: integration/cri-containerd/test.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -x
+set -eE
+
+err_report() {
+	cat /tmp/test-cri/cri-containerd.log
```
Comment (jodh-intel) [inline integration/cri-containerd/test.sh]: Looking at `hack/test-cri.sh`, this is the default, but we can change that by setting `REPORT_DIR`. I think we should do that as this directory may already exist for example. Alternatively, I'd delete `/tmp/test-cri/` before running the test.

File: integration/cri-containerd/test.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	cat /tmp/test-cri/containerd.log
+}
+
+trap 'err_report' ERR
+
+
+repo="github.com/containerd/cri-containerd"
+# FIXME: cri-containerd fails using macvtap.
+sudo sed -r -i -e 's/^(internetworking_model=).*/\1"bridged"/g' /usr/share/defaults/clear-containers/configuration.toml
```
Comment (jodh-intel) [inline integration/cri-containerd/test.sh]: Might be worth creating a variable for this path (atleast the `clear-containers/` element) to make it easier to migrate it to Kata?

File: integration/cri-containerd/test.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+CRITEST=${GOPATH}/bin/critest
+# make sure cri-containerd test install the proper installation for testing
+rm -f "${CRITEST}"
+pushd ${GOPATH}/src/${repo}
+
+FOCUS="runtime should support basic operations on container" CRI_CONTAINERD_FLAGS="--containerd-runtime-engine cc-runtime --stream-port 10011" make test-cri
```
Comment (jodh-intel) [inline integration/cri-containerd/test.sh]: Same comment as above - might be worth creating a variable for `cc-runtime`.

File: integration/cri-containerd/test.sh [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                TestContainerListStatsWithIdSandboxIdFilter
+                TestDuplicateName
+                TestImageLoad
+                TestImageFSInfo
+                TestSandboxCleanRemove
+             )
+
+for t in "${passing_test[@]}"
+do
+CRI_CONTAINERD_FLAGS="--containerd-runtime-engine cc-runtime --stream-port 10011" FOCUS="${t}" make test-integration
```
Comment (jodh-intel) [inline integration/cri-containerd/test.sh]: A runtime variable could be re-used here too.

File: integration/cri-containerd/test.sh [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+sudo sed -r -i -e 's/^(internetworking_model=).*/\1"bridged"/g' /usr/share/defaults/clear-containers/configuration.toml
+
+CRITEST=${GOPATH}/bin/critest
+# make sure cri-containerd test install the proper installation for testing
+rm -f "${CRITEST}"
+pushd ${GOPATH}/src/${repo}
+
+FOCUS="runtime should support basic operations on container" CRI_CONTAINERD_FLAGS="--containerd-runtime-engine cc-runtime --stream-port 10011" make test-cri
+
+passing_test=(
```
Comment (jodh-intel) [inline integration/cri-containerd/test.sh]: This is great. However, how about running all the tests and asserting that this list pass, and all the other tests fail?

Assuming we update `versions.txt` to pin the version of `cri-containerd` we're using, this should work as every time we update that version, and re-run the tests locally we *might* end up getting more tests passing, which can then be added to this list.

go1.9 provides a `go test -list '.*'` command which is supposed to list all tests in a package, but I cannot get it to work locally. As an alternative, this seems to be a more reliable method of listing all test files and test functions:

```
find . -type f|grep -v vendor/|grep "_test.go"|xargs grep "^func Test[^ ]"|sed 's/func //g'|cut -d\( -f1|cut -d: -f2|sort
```

File: integration/cri-containerd/test.sh [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+err_report() {
+	cat /tmp/test-cri/cri-containerd.log
+	cat /tmp/test-cri/containerd.log
+}
+
+trap 'err_report' ERR
+
+
+repo="github.com/containerd/cri-containerd"
+# FIXME: cri-containerd fails using macvtap.
```
Comment (devimc) [inline integration/cri-containerd/test.sh]: did you create an issue for this?

**Pull request closed without merging**